modules/core/src/main/java/io/fluo/core/impl/SharedResources.java

```diff
@@ -79,6 +81,14 @@ public CuratorFramework getCurator() {
     return curator;
   }
   
+  public synchronized OracleClient getOracleClient() {
+    checkIfClosed();
+    if (oracleClient == null) {
+      oracleClient = new OracleClient(env);
+    }
+    return oracleClient;
```

Comment:
I was thinking that OracleClient probably needed a close method but I am not too familiar with it.  Feel free to make a pull request.